packages/gatsby/cache-dir/navigation.js

```diff
@@ -159,6 +158,55 @@ function init() {
   maybeRedirect(window.location.pathname)
 }
 
+class RouteAnnouncer extends React.Component {
+  constructor(props) {
+    super(props)
+    this.announcementRef = React.createRef()
+  }
+
+  componentDidUpdate(prevProps, nextProps) {
+    requestAnimationFrame(() => {
+      let pageName = `new page at ${this.props.location.pathname}`
+      if (document.title) {
+        pageName = document.title
+      }
+      const pageHeadings = document
+        .getElementById(`gatsby-focus-wrapper`)
+        .getElementsByTagName(`h1`)
+      if (pageHeadings && pageHeadings.length) {
+        pageName = pageHeadings[0].textContent
+      }
+      const newAnnouncement = `Navigated to ${pageName}`
+      const oldAnnouncement = this.announcementRef.current.innerText
```

Comment:
Ah, one thing I just thought of. We should add a bailout check if the ref isn't attached yet. It feels unlikely to occur, but technically ref's type is `{current: HTMLElement | null}`. It's possible this could come up later on with suspense. All you need to do is at the top of this rAF, put a `if (!this.announcementRef.current) return`

Reply:
@blainekasten Suspense won’t change anything here.

It’s impossible for the ref to be null during componentDidUpdate, though note that because of the rAF call it is possible (though probably unlikely) for the component to unmount before this code runs, so a check is appropriate.

Reply:
@blainekasten I think this check is still missing in the code. Just talked to someone on Discord getting `Cannot read property 'innerText' of null` on this line. Don't know if that's the cause of his problem, or a side-effect.

Reply:
I'm getting this error while trying to debug problems with Ms Edge (14). Happens when I try to navigate to a new page.